azext_edge/edge/providers/support/base.py

```diff
@@ -183,13 +184,25 @@ def process_statefulset(
     statefulsets: V1StatefulSetList = v1_apps.list_stateful_set_for_all_namespaces(
         label_selector=label_selector, field_selector=field_selector
     )
+    namespace_pods_work = {}
 
-    return _process_kubernetes_resources(
+    processed = _process_kubernetes_resources(
         directory_path=directory_path,
         resources=statefulsets,
         kind=BundleResourceKind.statefulset.value,
     )
 
+    for statefulset in statefulsets.items:
+        statefulset_namespace: str = statefulset.metadata.namespace
+
+        if statefulset_namespace not in namespace_pods_work:
+            namespace_pods_work[statefulset_namespace] = True
+
+    if return_namespaces:
+        return processed, namespace_pods_work
```

Comment:
you should avoid mixing return types. the return type above is also no longer valid